### PR TITLE
Adds default to module.exports

### PR DESCRIPTION
**Pull request by fox1t**
Fixes: https://github.com/fastify/fastify-plugin/issues/105

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by zekth — APPROVED** [review]
LGTM

File: plugin.js [hunk 1]
```diff
@@ -122,4 +122,5 @@ function checkVersion (version, pluginName) {
   }
 }
 
+plugin.default = plugin
```
Comment (jsumners) [inline plugin.js:125]: What if the plugin already exports a `.default` property?

Comment (fox1t) [reply plugin.js:125]: This is `fastify-plugin` itself. The case of plugin exporting a default prop is already covered here: https://github.com/fastify/fastify-plugin/blob/master/plugin.js#L13

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by Eomm — COMMENTED** [review]
Would not be possible to add this bug case in the [plugin.test-d.ts](https://github.com/fastify/fastify-plugin/blob/master/plugin.test-d.ts) test file?

**Comment by fox1t** [comment]
Not really, because that case is already covered in that types tests. This is only testable if: 1. We port the codebase to TS (and this isn't an option); 2. Add tests written in TS instead of using tsd, that just checks types. 

I spoke about this issue long time ago with @delvedor: writing JS and testing with tsd doesn't cover the case where JS actually doesn't implements what the typings are saying.

**Comment by RafaelGSS** [comment]
> Not really, because that case is already covered in that types tests. This is only testable if: 1. We port the codebase to TS (and this isn't an option); 2. Add tests written in TS instead of using tsd, that just checks types.
> 
> I spoke about this issue long time ago with @delvedor: writing JS and testing with tsd doesn't cover the case where JS actually doesn't implements what the typings are saying.

If we use `ts-jest` and implement common usage like the reproduction code, could works?

**Comment by fox1t** [comment]
we don't beed another test suite, tap already covers our beds. If I remember correctly, we have other repository were we use it to run tests written in TS.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
> Add tests written in TS instead of using tsd, that just checks types.

Having a test written in ts with `tap` would be ok. tap supports ts via ts-node.

**Review by L2jLiga — APPROVED** [review]
👍

**Comment by fox1t** [comment]
I think we need to merge this asap because it is a really bad bug. This is borken for every TS user. After this merge we can open new PR to replace tsd with real TS tests.

**Review by zekth — APPROVED** [review]
LGTM

As @fox1t says we need to merge it now and we can fix the test runner in further PRs.

**Pull request merged**